python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py

```diff
@@ -270,44 +297,70 @@ async def on_messages(self, messages: Sequence[ChatMessage], cancellation_token:
         async for message in self.on_messages_stream(messages, cancellation_token):
             if isinstance(message, Response):
                 return message
-        raise AssertionError("The stream should have returned the final result.")
+        raise AssertionError(
+            "The stream should have returned the final result.")
 
     async def on_messages_stream(
         self, messages: Sequence[ChatMessage], cancellation_token: CancellationToken
     ) -> AsyncGenerator[AgentMessage | Response, None]:
+        # Query memory if available with the last message
+        memory_context = ""
+        if self._memory is not None and messages:
+            try:
+                last_message = messages[-1]
+                # ensure the last message is a text message or multimodal message
+                if not isinstance(last_message, TextMessage) and not isinstance(last_message, MultiModalMessage):
+                    raise ValueError(
+                        "Memory query failed: Last message must be a text message or multimodal message.")
+                results: List[MemoryQueryResult] = await self._memory.query(messages[-1].content, cancellation_token=cancellation_token)
```

Comment:
I guess this is saying that the default memory implementation is RAG with the last message. 

I feel that using the last message for the RAG is reasonable but restrictive, could we just pass messages and have the memory query decide? 

A high level comment that I'm not sure whether to call this memory or datastore/database.

Reply:
@husseinmozannar , I agree about passing the entire context and let the query method decide. 
```python
 query: ContentItem | List[ContentItem] 
 ``` 
 
I am flexible on naming. 
I like Memory because it connotes "just in time" **_retrieval/recall_** of content relevant to a step (in a task) an agent is about to take. Memory also gives a sense of **_what_** is stored inside the memory - in this case it really should be content relevant to task completion (not just anything that can be in a database).